components/server/src/workspace/config-provider.ts

```diff
@@ -258,12 +259,14 @@ export class ConfigProvider {
                 throw new Error(`Cannot fetch workspace image source for host: ${host}`);
             }
             const repoHost = hostContext.services;
-            const lastDockerFileSha = await repoHost.fileProvider.getLastChangeRevision(
-                repository,
-                revisionOrTagOrBranch,
-                user,
-                dockerFilePath,
-            );
+            const lastDockerFileSha = await repoHost.fileProvider
+                .getLastChangeRevision(repository, revisionOrTagOrBranch, user, dockerFilePath)
+                .catch((e) => {
+                    if (e instanceof RevisionNotFoundError) {
+                        return "";
```

Comment:
This is fairly important. We now have this additional null string value when the Dockerfile has not been found. This is done so that we don't have to re-type everything, and it also means that when the incremental ws service resolves the image, it always considers it fresh (since no valid prebuild will have a missing Dockerfile)

Reply:
Also, you could make it a constant and use that. Also, it could be another "special" value, that maybe is a bit more readable if future us is stumbling over this...?

Reply:
Addressed in https://github.com/gitpod-io/gitpod/pull/20345#discussion_r1832457182 🙏.